### PR TITLE
`userProfile` removed from `Global` singleton

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

This PR is a series of commits leading to removal of `userProfile` property from `Global.qml`. It's part of broader effort aiming making singletons stateless and backend-independent.

- `isDisplayNameDupeOfCommunityMember` moved from `Utils` singleton to `CommunitiesStore` as it calls backend
- display name validators moved from `Constants` to a dedicated component, explicitly dependent on a store. It makes `Constant` holding really read-only, immutable data, with no references to the backend.
- `ContactsStore.getProfileContext` simplified, no need to provide user's own public key
- `UserStatusContextMenu` made purely UI component, actions handling externalized
- `ProfileHeader` made backend-independent and behavior improved - after removing icon (in a dirty state) the menu entry for removing image is not available (earlier was available and did nothing when clicked)
- `sendModalPopup` removed from `ProfileSectionStore`
- several direct references to `userProfile` context property removed from UI code
- unused `PopupMenu` removed

Closes: #16459 

### Affected areas
Multiple components, mainly those using `Global.userProfile`

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: ui/app/AppLayouts/Profile/panels/ProfileDescriptionPanel.qml [hunk 1]
```diff
@@ -33,7 +33,6 @@ Item {
             label: qsTr("Display name")
             placeholderText: qsTr("Display Name")
             charLimit: Constants.keypair.nameLengthMax
-            validators: Constants.validators.displayName
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/panels/ProfileDescriptionPanel.qml:36]: Why is this removed? :thinking:

**Review by caybro — COMMENTED** [review]
LGTM otherwise

File: ui/app/AppLayouts/Profile/panels/ProfileDescriptionPanel.qml (hunk 1 above)
Comment (micieslak) [reply ui/app/AppLayouts/Profile/panels/ProfileDescriptionPanel.qml:36]: Because the same thing (with some other conditions) is set in `ui/app/AppLayouts/Profile/views/MyProfileView.qml`

**Review by caybro — APPROVED** [review]
LGTM

**Review by jrainville — APPROVED** [review]
LGTM
